prusaerrors/sl1/codes.py

```diff
@@ -10,15 +10,72 @@
 
 import builtins
 from pathlib import Path
+from re import compile as re_compile
 
-from prusaerrors.shared.codes import unique_codes, Codes, yaml_codes
+import yaml
+
+from prusaerrors.shared.codes import Category, Code, Printer, unique_codes, Codes
+from prusaerrors.sl1 import PRINTER_MODEL_PATH
 
 if "_" not in vars(builtins):
 
     def _(value):
         return value
 
 
+def yaml_codes(src_path: Path):
+    """
+    Add code definitions from YAML source
+    """
+
+    def decor(cls):
+        with src_path.open("r") as src_file:
+            data = yaml.safe_load(src_file)
+        assert "Errors" in data
+
+        printer = Printer.SL1
+        model = [x.name for x in Path(str(PRINTER_MODEL_PATH)).iterdir() if x.is_file()]
+        if len(model) == 1:
+            a = Printer.M1.name.lower()
+            if a in model:
+                printer = Printer.M1
+        else:
+            raise KeyError(f"None or multiple model files found. Check {PRINTER_MODEL_PATH} folder.")
```

Comment:
```suggestion
        if len(model) != 1:
            raise KeyError(f"None or multiple model files found. Check {PRINTER_MODEL_PATH} folder.")
        model = model[0]
 
        for p in Printer:
            if p.value.name.lower() == model:
                printer = p

        if not printer:
            raise KeyError(f"Unknown printer model: \"{model}\". Check {PRINTER_MODEL_PATH} folder.")
```
This change + initialize printer as None should ensure unknown keyword is not detected as SL1.

Reply:
Well I couldn't find how to properly mock Path(). Its fixed now.